lib/rspec/mocks/proxy.rb

```diff
@@ -31,6 +33,25 @@ def initialize(object, order_group, options={})
         @method_doubles = Hash.new { |h, k| h[k] = MethodDouble.new(@object, k, self) }
       end
 
+      # @private
+      def ensure_can_be_proxied!(object)
+        if RUBY_VERSION.to_f < 2.0
+          return if !object.is_a?(Symbol) && !object.frozen?
+        else
+          return unless object.frozen?
+        end
```

Comment:
I'm ok with this check all the time to be honest.
```suggestion
        return unless object.is_a?(Symbol) || object.frozen?
```

Reply:
applied your variant